utils/wasm-gen/src/generator/syscalls/imports.rs

```diff
@@ -242,36 +242,92 @@ impl<'a, 'b> SysCallsImportsGenerator<'a, 'b> {
 }
 
 impl<'a, 'b> SysCallsImportsGenerator<'a, 'b> {
-    /// Generates a function which calls "properly" the `gr_reservation_send`.
-    fn generate_send_from_reservation(&mut self) {
-        let (reserve_gas_idx, reservation_send_idx) = {
-            let maybe_reserve_gas = self
-                .sys_calls_imports
-                .get(&InvocableSysCall::Loose(SysCallName::ReserveGas))
-                .map(|&(_, call_indexes_handle)| call_indexes_handle);
-            let maybe_reservation_send = self
-                .sys_calls_imports
-                .get(&InvocableSysCall::Loose(SysCallName::ReservationSend))
-                .map(|&(_, call_indexes_handle)| call_indexes_handle);
-
-            match maybe_reserve_gas.zip(maybe_reservation_send) {
-                Some(indexes) => indexes,
+    /// Returns the indexes of invocable sys-calls.
+    fn invocable_sys_calls_indexes<const N: usize>(
+        &self,
+        sys_call: SysCallName,
+        sys_calls: [SysCallName; N],
+    ) -> Option<[usize; N]> {
+        let mut indexes = [0; N];
+        let iter = sys_calls.iter().map(|&sys_call| {
+            self.sys_calls_imports
+                .get(&InvocableSysCall::Loose(sys_call))
+                .map(|&(_, call_indexes_handle)| call_indexes_handle)
+        });
+
+        for (index, maybe_index) in indexes.iter_mut().zip(iter) {
+            match maybe_index {
+                Some(idx) => *index = idx,
                 None => {
-                    log::trace!("Wasm hasn't got either gr_reserve_gas, or gr_reservation_send, or both of them");
-                    return;
+                    log::trace!(
+                        "To build the {precise_sys_call_name} sys-call, wasm must import the following sys-calls: {missing_sys_calls}",
+                        precise_sys_call_name = InvocableSysCall::Precise(sys_call).to_str(),
+                        missing_sys_calls = sys_calls
+                            .iter()
+                            .map(|sys_call| sys_call.to_str())
+                            .collect::<Vec<_>>()
```

Comment:
This allocation is really odd. Just create a separate converter function in `utlis.rs`